src/lightmap.cpp

```diff
@@ -651,6 +652,14 @@ float map::light_transparency( const tripoint &p ) const
     return get_cache_ref( p.z ).transparency_cache[p.x][p.y];
 }
 
+void map::get_lightmap( const int zlev )
+{
+    if( level_cache::lightmap_latest_zlevel == zlev ) {
+        return;
+    }
+    return generate_lightmap( zlev );
```

Comment:
This still calls `map::build_sunlight_cache()` which overwrites at least one of the caches above zlev with just sunlight.